akka-http-core/src/main/scala/akka/http/scaladsl/settings/Http2ServerSettings.scala

```diff
@@ -26,6 +29,24 @@ private[http] trait Http2CommonSettings {
   def logFrames: Boolean
   def maxConcurrentStreams: Int
   def outgoingControlFrameBufferSize: Int
+
+  def pingInterval: FiniteDuration
+  def pingTimeout: FiniteDuration
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[http] object Http2CommonSettings {
+  def validate(settings: Http2CommonSettings): Unit = {
+    import settings._
+    if (pingInterval.toSeconds > 0) {
+      require(pingInterval.toSeconds.seconds == pingInterval, s"ping-interval must be whole seconds, was $pingInterval")
+      require(pingTimeout == 0.seconds || pingTimeout < pingInterval, "ping-timeout must be zero, or positive and smaller than ping-interval")
```

Comment:
```suggestion
      require(pingTimeout == 0.seconds || (pingTimeout > 0.seconds && pingTimeout < pingInterval), "ping-timeout must be zero, or positive and smaller than ping-interval")
```
?

Reply:
I think the case for `0s` disabling the timeout checks is not documented in `reference.conf` and would break the condition that only one PING can be in-flight at any given time.

Reply:
I'm leaving that for now since I'm looking at implementing a non static tick interval which changes this a bit.

Reply:
It's tricky because as soon as we leave the possibility to send two pings without first seeing an ack, we don't know which ack we got back and the protocol requires the other side to respond etc. Given the value of the feature I'd prefer to not make this more complicated by having several outstanding pings and keeping track of them. In reality I think you will likely set a ping interval of a minute or higher and the responses will either come in milliseconds or not at all.